File: event/src/test/java/de/danoeh/antennapod/event/settings/SpeedPresetChangedEventTest.java
package de.danoeh.antennapod.event.settings;

import org.junit.Test;

import de.danoeh.antennapod.model.feed.FeedPreferences;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertSame;

public class SpeedPresetChangedEventTest {

    @Test
    public void speedFeedIdAndSkipSilenceAreStored() {
        SpeedPresetChangedEvent event =
                new SpeedPresetChangedEvent(1.25f, 11L, FeedPreferences.SkipSilence.AGGRESSIVE);
        assertEquals(1.25f, event.getSpeed(), 0f);
        assertEquals(11L, event.getFeedId());
        assertSame(FeedPreferences.SkipSilence.AGGRESSIVE, event.getSkipSilence());
    }
}

File: event/src/test/java/de/danoeh/antennapod/event/settings/SpeedPresetChangedEventTest.kt
package de.danoeh.antennapod.event.settings

import de.danoeh.antennapod.model.feed.FeedPreferences
import org.junit.Assert.assertEquals
import org.junit.Assert.assertSame
import org.junit.Test

class SpeedPresetChangedEventTest {

    @Test
    fun speedFeedIdAndSkipSilenceAreStored() {
        val event =
            SpeedPresetChangedEvent(1.25f, 11L, FeedPreferences.SkipSilence.AGGRESSIVE)
        assertEquals(1.25f, event.speed, 0f)
        assertEquals(11L, event.feedId)
        assertSame(FeedPreferences.SkipSilence.AGGRESSIVE, event.skipSilence)
    }
}
